crates/bevy_pbr/src/render/pbr_functions.wgsl

```diff
@@ -268,3 +269,16 @@ fn dither(color: vec4<f32>, pos: vec2<f32>) -> vec4<f32> {
 }
 #endif
 
+fn premultiply_alpha(standard_material_flags: u32, color: vec4<f32>) -> vec4<f32> {
+    let alpha_mode = standard_material_flags & STANDARD_MATERIAL_FLAGS_ALPHA_MODE_BITS;
+    if (alpha_mode == STANDARD_MATERIAL_FLAGS_ALPHA_MODE_BLEND) {
+        return vec4<f32>(color.rgb * color.a, color.a);
```

Comment:
Is this correct? I'm not sure standard blending should multiple the color by the alpha. 🤔

Reply:
It's counterintuitive, but I believe it is correct:

As suggested by (I believe?) @fintelia, to avoid the inefficiency of having three separate pipelines, the `Blend`, `Premultiplied` and `Add` modes all share the same pipeline key (`MeshPipelineKey::BLEND_PREMULTIPLIED_ALPHA`) which produces the same blend state (`BlendState::PREMULTIPLIED_ALPHA_BLENDING`) — changed from `BlendState::ALPHA_BLENDING` that we used before.

This mode assumes alpha is premultiplied, and depending on what we do with the alpha channel and the color we get different results:

- If we multiply the color and keep the alpha channel, we restore the previous behavior we had under `BlendState::ALPHA_BLENDING` (used for `Blend`)
- If we multiply the color but change the alpha channel to `0.0`, we get additive blending
- If we don't change the color at all, we get the normal/expected behavior of `BlendState::PREMULTIPLIED_ALPHA_BLENDING`.

Reply:
Added some comments clarifying the math to the shader code.